### PR TITLE
Model method improvements

### PR DESCRIPTION
**Pull request by mikez321**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling (Styling to new or existing files)
- [X] Testing
### Why is this change required? What problem does it solve?
-Plant search now returns results alphabetically
-Random plant endpoint pulls random plant from DB, not generating a random ID.  This fixes an issue where an ID could be created for a plant that had been removed form the DB and would potentially cause the system to break.
### Was any previously written code changed?
Removed plant model method of generate random ID and replaced it with get random plant.  Also added an order by line to the plant search method.

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -24,15 +25,12 @@ def get_by_id(id):
...
         return plant
 
-    def random_id():
-        from random import randint
-        plants = Plant.query.all()
-        max = len(plants)
-        num = randint(1, max)
-        return num
+    def random_plant():
+        plant = Plant.query.order_by(func.random()).first()
```
Comment (kelshadarby) [inline app/models.py:29]: Love this method!

File: app/models.py [hunk 2]
```diff
@@ -24,15 +25,12 @@ def get_by_id(id):
...
+        return plant
 
     def plant_search(search):
-        plants = db.session.query(Plant).filter(Plant.plant_type.ilike('%{}%'.format(search))).all()
+        plants = db.session.query(Plant).filter(Plant.plant_type.ilike('%{}%'.format(search))).order_by(Plant.plant_type).all()
```
Comment (kelshadarby) [inline app/models.py:33]: I like this ordering

File: app/models.py [hunk 3]
```diff
@@ -2,6 +2,7 @@
 from datetime import datetime
 from flask_sqlalchemy import SQLAlchemy
 from sqlalchemy.orm import relationship, backref
+from sqlalchemy import func
```
Comment (kelshadarby) [inline app/models.py:5]: IMPORT THE FUNK 🤠🥳
funky gif

File: tests/test_endpoint.py [hunk 4]
```diff
@@ -116,6 +116,25 @@ def test_api_can_search_for_a_plant_type(self):
...
+        jimmy = Plant(plant_type='Cherry Tomato',image='jim_photo.jpg',lighting='Full Sun',water_frequency=3,harvest_time=50,root_depth=12,annual="Annual")
+        agatha = Plant(plant_type='Roma Tomato',image='agatha_photo.jpg',lighting='Full Sun',water_frequency=2,harvest_time=60,root_depth=12,annual="Annual")
+        dan = Plant(plant_type='Cactus',image='cactus_dan.jpg',lighting='Full Sun',water_frequency=7,harvest_time=None,root_depth=8,annual="Annual")
+        db.session.add_all([jimmy, agatha, dan])
+        db.session.commit()
+        res = self.client().get('/api/v1/plants/search?q=a')
+        self.assertEqual(res.status_code, 200)
+
+        json_response = json.loads(res.data)
+        # For easier readability, each plant type is listed in order below:
```
Comment (kelshadarby) [inline tests/test_endpoint.py:129]: This makes it very readable for anyone checking this out in the future

File: app/models.py (hunk 3 above)
Comment (mikez321) [reply app/models.py:5]: Amazing!

**Pull request merged**